### PR TITLE
Created Recipe API

### PR DESCRIPTION
**Pull request by hwbit**
Recipe API and updated Recipe schema.

GET `/api/recipes/` - returns all recipes
GET `/api/recipes/lookupId/:id` - get recipe by ID
GET `/api/recipes/lookupAuthor/:author` - get all recipes by an author
POST `/api/recipes/searchName` - used for search since people may have spaces
POST `/api/recipes/createRecipe` - create a recipe
POST `/api/recipes/updateRecipe` - update a recipe
DELETE `/api/recipes/deleteRecipe/:id` - delete a recipe

### CODE REVIEW

File: preppal-be/routes/recipeApi.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+});
+
+/**
+ * POST - Update recipe
+ */
+routerRecipeApi.post("/updateRecipe", async (req, res) => {
+    try {
+        const { _id, author, recipeTitle, recipeTitleUrl, description, ingredients, instructions, servingSize, prepTime, cookingTime, creationDate } = req.body;
+        const modifiedDate = Date.now();
```
[the remark this answers is not in the document]
Comment (hwbit) [reply preppal-be/routes/recipeApi.ts]: fixed

File: preppal-be/models/recipe.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-    },
     author: {
         type: String,
         required: true
     },
+    recipeTitle: {
+        type: String,
+        required: true
+    },
+    recipeTitleUrl: {
```
[the remark this answers is not in the document]
Comment (hwbit) [reply preppal-be/models/recipe.ts:16]: added a message for its intentions

File: preppal-be/models/recipe.ts [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
 const RecipeSchema = new Schema({
-    recipeId: {
-        type: UUID,
-        required: UUID.generate()
-    },
     author: {
         type: String,
         required: true
     },
+    recipeTitle: {
```
Comment (lik5) [inline preppal-be/models/recipe.ts:11]: we should change this to just title

File: preppal-be/models/recipe.ts [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
+        type: String,
+        required: true
+    },
+    // TODO Use this for URL address encoding
+    recipeTitleUrl: {
```
Comment (lik5) [inline preppal-be/models/recipe.ts:16]: change this one as well

File: preppal-be/models/recipe.ts [hunk 5]
```diff
@@ -1,16 +1,22 @@
-import { UUID } from "mongodb";
 const mongoose = require('mongoose');
 const Schema = mongoose.Schema;
 
+const date = new Date().toString();
+
 const RecipeSchema = new Schema({
```
Comment (lik5) [inline preppal-be/models/recipe.ts:6]: should we have something for images?

Comment (hwbit) [reply preppal-be/models/recipe.ts:6]: added image to schema, it will take the url address.

**Review by lik5 — APPROVED** [review]

**Pull request merged**